src/main/java/org/testng/internal/Parameters.java

```diff
@@ -52,11 +52,39 @@
       AfterMethod.class
   };
 
-  private static Map<String, List<Class>> mapping = Maps.newHashMap();
+  private static Map<String, List<Class<?>>> mapping = Maps.newHashMap();
+/*
+          +--------------+--------------+---------+--------+----------+-------------+
+          |  Annotation  | ITestContext | XmlTest | Method | Object[] | ITestResult |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | BeforeSuite  | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | BeforeTest   | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | BeforeGroups | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | BeforeClass  | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | BeforeMethod | Yes          | Yes     | Yes    | Yes      | Yes         |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | AfterSuite   | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | AfterTest    | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | AfterGroups  | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | AfterClass   | Yes          | Yes     | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | AfterMethod  | Yes          | Yes     | Yes    | Yes      | Yes         |
+          +--------------+--------------+---------+--------+----------+-------------+
+          | Test         | Yes          | No      | No     | No       | No          |
+          +--------------+--------------+---------+--------+----------+-------------+
 
+ */
   static {
-    List<Class> ctxTest = Arrays.asList(new Class[] {ITestContext.class, XmlTest.class});
-    List<Class> mParams = Arrays.asList(new Class[] {ITestContext.class, XmlTest.class,Method.class,Object[].class});
+    List<Class<?>> ctxTest = Arrays.asList(new Class<?>[]{ITestContext.class, XmlTest.class});
```

Comment:
@juherr / @cbeust - I raised a PR to the documentation site https://github.com/testng-team/testng-team.github.io/pull/20 can you please help take a look ?